main.go

```diff
@@ -78,12 +80,8 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = (&xgboostcontroller.XGBoostJobReconciler{
-		Client: mgr.GetClient(),
-		Log:    ctrl.Log.WithName("controllers").WithName("XGBoostJob"),
-		Scheme: mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "XGBoostJob")
+	if err = pytorchcontroller.NewReconciler(mgr).SetupWithManager(mgr); err != nil {
```

Comment:
it might have some conflicts with #1293 after the merge. I can handle the conflicts.